src/profile-logic/call-node-info.js

```diff
@@ -331,21 +783,528 @@ export class CallNodeInfoInvertedImpl
+    const firstChildFunc = callNodeTable.func[firstChildFirstDeepNode];
+
+    const deepNodeCountPerChild = [];
+    const funcPerChild = [];
+
+    let currentChildFunc = firstChildFunc;
+    let currentChildDeepNodeCount = 1;
+    for (let j = 1; j < childrenDeepNodeCount; j++) {
+      const parentDeepNode = parentDeepNodes[nodesWhichEndHereCount + j];
+      const deepNode = callNodeTable.prefix[parentDeepNode];
+      childrenDeepNodes[j] = deepNode;
+      // assert(deepNode !== -1, "parentDeepNodes is sorted so that all call paths which end at this depth come first (by definition of the suffix order), and we already skipped those");
+      const deepNodeFunc = callNodeTable.func[deepNode];
+      // assert(currentChildFunc <= deepNodeFunc, "parentDeepNodes is sorted by prefix func, by definition of the suffix order (at least in this range, because the rest of the call path is identical for all nodes in parentDeepNodes)");
+
+      if (deepNodeFunc !== currentChildFunc) {
+        funcPerChild.push(currentChildFunc);
+        deepNodeCountPerChild.push(currentChildDeepNodeCount);
+        currentChildFunc = deepNodeFunc;
+        currentChildDeepNodeCount = 0;
+      }
+      currentChildDeepNodeCount++;
+    }
+    funcPerChild.push(currentChildFunc);
+    deepNodeCountPerChild.push(currentChildDeepNodeCount);
+
+    const childrenSuffixOrderIndexRangeStart =
+      parentIndexRangeStart + nodesWhichEndHereCount;
+
+    return {
+      funcPerChild,
+      deepNodeCountPerChild,
+      childrenSuffixOrderIndexRangeStart,
+      childrenDeepNodes,
+    };
+  }
+
+  /**
+   * Create the children for parentNodeHandle based on the information in
+   * childrenInfo.
+   *
+   * Returns the handles of the created children. The returned array is ordered
+   * by func.
+   */
+  _createChildrenForInfo(
+    childrenInfo: ChildrenInfo,
+    parentNodeHandle: InvertedCallNodeHandle
+  ): InvertedCallNodeHandle[] {
+    const parentNodeCallPathHash = this._pathHashForNode(parentNodeHandle);
+    const childrenDepth = this.depthForNode(parentNodeHandle) + 1;
+
+    const {
+      funcPerChild,
+      deepNodeCountPerChild,
+      childrenSuffixOrderIndexRangeStart,
+      childrenDeepNodes,
+    } = childrenInfo;
+
+    const childCount = funcPerChild.length;
+    const childCallNodes = [];
+    let nextChildDeepNodeIndex = 0;
+    let nextChildSuffixOrderIndexRangeStart =
+      childrenSuffixOrderIndexRangeStart;
+    for (let childIndex = 0; childIndex < childCount; childIndex++) {
+      const func = funcPerChild[childIndex];
+      const deepNodeCount = deepNodeCountPerChild[childIndex];
+
+      const suffixOrderIndexRangeStart = nextChildSuffixOrderIndexRangeStart;
+      const childDeepNodes = childrenDeepNodes.subarray(
+        nextChildDeepNodeIndex,
+        nextChildDeepNodeIndex + deepNodeCount
+      );
+      nextChildSuffixOrderIndexRangeStart += deepNodeCount;
+      nextChildDeepNodeIndex += deepNodeCount;
```

Comment:
(optional) if you move these additions at the end of the loop, I think this could be somewhat clearer (it's common that the end of a loop contains the operations for the next iteration), also I think you could rename them `currentChildXXX` instead of `next`, and use them directly (so you could remove `suffixOrderIndexRangeStart` too).